server/services/assistant_client.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ApiResponse } from '@opensearch-project/opensearch';
+
+import {
+  OpenSearchClient,
+  OpenSearchDashboardsRequest,
+  RequestHandlerContext,
+} from '../../../../src/core/server';
+import { ML_COMMONS_BASE_API } from '../utils/constants';
+import { getAgent } from '../routes/get_agent';
+
+interface AgentExecuteResponse {
+  inference_results: Array<{
+    output: Array<{ result: string }>;
+  }>;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const isDataSourceQuery = (query: any): query is { dataSourceId: string } => {
+  if ('dataSourceId' in query && query.dataSourceId) {
+    return true;
+  }
+  return false;
+};
+
+export class AssistantClient {
+  constructor(
+    private request: OpenSearchDashboardsRequest,
+    private context: RequestHandlerContext & {
+      dataSource?: {
+        opensearch: {
+          getClient: (dataSourceId: string) => Promise<OpenSearchClient>;
+        };
+      };
+    }
+  ) {}
+
+  executeAgent = async (
+    agentName: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    parameters: Record<string, any>
```

Comment:
do we want to support executeAgent with agent id?

Reply:
I also considered this, but we don't have use case at the moment which requires to execute agent with id directly, so I decided to not add it for now, we can add it when it's needed.

Reply:
+1

Reply:
Knowledge base needs to execute by agent id directly

Reply:
@qianheng-aws may have a requirement to execute agent directly

Reply:
@Hailong-am @qianheng-aws Updated, could you take another look?

Reply:
looks good to me.

Reply:
LGTM